test-framework/core/src/main/java/org/keycloak/test/framework/LogFilter.java

```diff
@@ -0,0 +1,29 @@
+package org.keycloak.test.framework;
+
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.logging.Filter;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+public class LogFilter implements Filter {
+
+    private final Queue<LogRecord> queue = new LinkedList<>();
+
+    @Override
+    public boolean isLoggable(LogRecord record) {
+        queue.add(record);
+        return false;
```

Comment:
@stianst Is it supposed to be every time false?

Reply:
Yer, so the way it works if you look at LogHandler is that it installs the filter, which queues all the log statements, then once the test finishes it uninstalls the filter, and depending on the status it clears the queued log statements, or logs them.

Ideally I would have added a single instance of the filter, with some on/off flag, but that doesn't work all that well since Quarkus can end up re-configuring logging whenever the embedded server is started. That's why it just adds/removes the filter.

Reply:
thanks for explanation :+1: